internal/mode/static/nginx/modules/src/httpmatches.js

```diff
@@ -50,22 +50,26 @@ function redirect(r) {
 }
 
 function extractMatchesFromRequest(r) {
-  if (!r.variables[MATCHES_VARIABLE]) {
-    throw Error(
-      `cannot redirect the request; the variable ${MATCHES_VARIABLE} is not defined on the request object`,
-    );
-  }
-
+  let index = 0;
   let matches;
 
-  try {
-    matches = JSON.parse(r.variables[MATCHES_VARIABLE]);
-  } catch (e) {
-    throw Error(
-      `cannot redirect the request; error parsing ${r.variables[MATCHES_VARIABLE]} into a JSON object: ${e}`,
-    );
+  while (true) {
```

Comment:
Is this loop just overwriting the `matches` var every time? So when you look for a match, you're only looking at the last iteration.